tests/connect/test_io.py

```diff
@@ -4,88 +4,45 @@
 
 import pytest
 
+import daft
 
-def test_write_csv_basic(spark_session, tmp_path):
-    df = spark_session.range(10)
+
+def test_csv_basic_roundtrip(make_spark_df, assert_spark_equals, spark_session, tmp_path):
+    df = make_spark_df({"id": [1, 2, 3]})
     csv_dir = os.path.join(tmp_path, "csv")
     df.write.csv(csv_dir)
 
-    csv_files = [f for f in os.listdir(csv_dir) if f.endswith(".csv")]
-    assert len(csv_files) > 0, "Expected at least one CSV file to be written"
-
-    df_read = spark_session.read.csv(str(csv_dir))
-    df_pandas = df.toPandas()
-    df_read_pandas = df_read.toPandas()
-    assert df_pandas["id"].equals(df_read_pandas["id"]), "Data should be unchanged after write/read"
-
-
-def test_write_csv_with_header(spark_session, tmp_path):
-    df = spark_session.range(10)
-    csv_dir = os.path.join(tmp_path, "csv")
-    df.write.option("header", True).csv(csv_dir)
+    spark_df_read = spark_session.read.option("header", True).csv(csv_dir)
+    print(spark_df_read.collect())
+    df_read = daft.read_csv(csv_dir)
+    print(df_read.collect())
+    assert_spark_equals(df_read, spark_df_read)
 
-    df_read = spark_session.read.option("header", True).csv(str(csv_dir))
-    df_pandas = df.toPandas()
-    df_read_pandas = df_read.toPandas()
-    assert df_pandas["id"].equals(df_read_pandas["id"])
 
+@pytest.mark.skip(reason="https://github.com/Eventual-Inc/Daft/issues/3775")
+def test_write_csv_with_header(make_df, make_spark_df, spark_session, tmp_path):
+    pass
```

Comment:
nit: based off the issue, I am assuming this is supposed to be without a header

```suggestion
@pytest.mark.skip(reason="https://github.com/Eventual-Inc/Daft/issues/3775")
def test_write_csv_without_header(make_df, make_spark_df, spark_session, tmp_path):
    pass
```